From 1b95c833d8cda62bf79f0ec8da7cca8f3ea1e4d8 Mon Sep 17 00:00:00 2001
From: topjohnwu <topjohnwu@gmail.com>
Date: Sun, 14 Jul 2024 16:55:01 -0700
Subject: [PATCH 06/17] Add missing sigisemptyset implementation

---
 shell/hush.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/shell/hush.c b/shell/hush.c
index d111f0cc5..5fbceaae3 100644
--- a/shell/hush.c
+++ b/shell/hush.c
@@ -1942,6 +1942,15 @@ enum {
 		,
 };
 
+#ifdef __BIONIC__
+static int sigisemptyset(const sigset_t *set)
+{
+    sigset_t empty;
+    sigemptyset(&empty);
+    return memcmp(&empty, set, sizeof(sigset_t)) == 0;
+}
+#endif
+
 static void record_pending_signo(int sig)
 {
 	sigaddset(&G.pending_set, sig);
-- 
2.45.2

