From 96a4a2663497869af67a842b35fa97388068c36b Mon Sep 17 00:00:00 2001
From: topjohnwu <topjohnwu@gmail.com>
Date: Fri, 3 Apr 2020 02:09:55 -0700
Subject: [PATCH 30/34] Fix incorrect assumptions about NDK

---
 include/platform.h       |  4 +++-
 libbb/missing_syscalls.c | 15 ---------------
 2 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/include/platform.h b/include/platform.h
index 57f0ef57d..69687eca7 100644
--- a/include/platform.h
+++ b/include/platform.h
@@ -549,8 +549,10 @@ typedef unsigned smalluint;
 # if __ANDROID_API__ >= 21
 #  undef HAVE_WAIT3
 # endif
+# if __ANDROID_API__ < 24
+#  undef HAVE_STRCHRNUL
+# endif
 # undef HAVE_MEMPCPY
-# undef HAVE_STRCHRNUL
 # undef HAVE_STRVERSCMP
 # undef HAVE_UNLOCKED_LINE_OPS
 # undef HAVE_NET_ETHERNET_H
diff --git a/libbb/missing_syscalls.c b/libbb/missing_syscalls.c
index f02e8d14c..a6db67da9 100644
--- a/libbb/missing_syscalls.c
+++ b/libbb/missing_syscalls.c
@@ -54,21 +54,6 @@
 #if defined(ANDROID) || defined(__ANDROID__)
 /*# include <linux/timex.h> - for struct timex, but may collide with <time.h> */
 # include <sys/syscall.h>
-pid_t getsid(pid_t pid)
-{
-	return syscall(__NR_getsid, pid);
-}
-
-int sethostname(const char *name, size_t len)
-{
-	return syscall(__NR_sethostname, name, len);
-}
-
-struct timex;
-int adjtimex(struct timex *buf)
-{
-	return syscall(__NR_adjtimex, buf);
-}
 
 int pivot_root(const char *new_root, const char *put_old)
 {
-- 
2.34.1

