From: John Wu <topjohnwu@gmail.com>
Date: Sun, 12 Aug 2018 18:31:26 +0800
Subject: [PATCH] platform: minor adjustments to support unified headers

---
 include/platform.h                 |  2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/platform.h b/include/platform.h
index 9e9ffa782..f813b956c 100644
--- a/include/platform.h
+++ b/include/platform.h
@@ -120,7 +127,7 @@
  * (-DFAST_FUNC= )
  */
 #ifndef FAST_FUNC
-# if __GNUC_PREREQ(3,0) && defined(i386)
+# if !defined(__ANDROID__) && __GNUC_PREREQ(3,0) && defined(i386)
 /* stdcall makes callee to pop arguments from stack, not caller */
 #  define FAST_FUNC __attribute__((regparm(3),stdcall))
 /* #elif ... - add your favorite arch today! */
