From: John Wu <topjohnwu@gmail.com>
Date: Sun, 12 Aug 2018 18:31:26 +0800
Subject: [PATCH] glob: minor adjustments to support unified headers

---
 shell/glob.h                       |  6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/shell/glob.h b/shell/glob.h
index e8e65786d..c274a8f6a 100644
--- a/shell/glob.h
+++ b/shell/glob.h
@@ -47,7 +47,7 @@
 #include <sys/cdefs.h>
 //#include <sys/_types.h>
 
-#ifndef	_SIZE_T_DECLARED
+#if !defined(_SIZE_T_DECLARED) && !defined(__ANDROID__)
 typedef	__size_t	size_t;
 #define	_SIZE_T_DECLARED
 #endif
@@ -74,7 +74,7 @@ typedef struct {
 	int (*gl_stat)(const char *, struct stat *);
 } glob_t;
 
-#if __POSIX_VISIBLE >= 199209
+#if __POSIX_VISIBLE >= 199209 || defined(__ANDROID__)
 /* Believed to have been introduced in 1003.2-1992 */
 #define	GLOB_APPEND	0x0001	/* Append to output from previous call. */
 #define	GLOB_DOOFFS	0x0002	/* Use gl_offs. */
@@ -91,7 +91,7 @@ typedef struct {
 #define	GLOB_NOSYS	(-4)	/* Obsolete: source comptability only. */
 #endif /* __POSIX_VISIBLE >= 199209 */
 
-#if __BSD_VISIBLE
+#if __BSD_VISIBLE || defined(__ANDROID__)
 #define	GLOB_ALTDIRFUNC	0x0040	/* Use alternately specified directory funcs. */
 #define	GLOB_BRACE	0x0080	/* Expand braces ala csh. */
 #define	GLOB_MAGCHAR	0x0100	/* Pattern had globbing characters. */
